File: Python WorldSimulatorGame/Mlecz.py
import random

from Roslina import Roslina


class Mlecz(Roslina):

    def __init__(self, polozenie, swiat):
        super(Mlecz, self).__init__(0, 0, 'M', 0, True, polozenie, swiat)

    def getNazwaOrganizmu(self):
        return "Mlecz"

    def akcja(self):
        self.wiek += 1
        self.ogolneInformacje()
        i = 0
        while True:
            i += 1
            x = random.randint(0, 9)
            if x == 0:
                self.rozprzestrzenianie()
                break
            if i == 3:
                break

File: C++ WorldSimulatorGame/POproj1/Mlecz.cpp
#include "Mlecz.h"

Mlecz::Mlecz(Punkt polozenie, Swiat* swiat)
	:Roslina(0, 0, 'M', 0, true, Organizmy::mlecz, polozenie, swiat) {
	this->sila = 0;
	this->inicjatywa = 0;
	this->znak = 'M';
	this->wiek = 0;
}
string Mlecz::getNazwaOrganizmu() {
	return "Mlecz";
}

void Mlecz::akcja() {
	this->wiek++;
	ogolneInformacje();
	int i = 0;
	//cout << this->getNazwaOrganizmu() << " ";
	while (true) {
		i++;
		int x = rand() % 10;
		if (x < 1) {
			Rozprzestrzenianie();
			//cout << "Rozprzestrzenia sie!" << endl;
			break;
		}
		if (i == 3) {
			i++;
			break;
		}
	}
	if (i == 4) {
		swiat->info.dodajInfocjmacje(" Nie udalo sie rozprzestrzenic " );
		//cout << " Nie udalo sie rozprzestrzenic " << endl;
	}
	swiat->info.dodajInfocjmacje("\n");
}